subsys/llext/llext_mem.c

```diff
@@ -28,10 +28,22 @@ LOG_MODULE_DECLARE(llext, CONFIG_LLEXT_LOG_LEVEL);
 #endif
 
 #ifdef CONFIG_LLEXT_HEAP_DYNAMIC
+#ifdef CONFIG_HARVARD
+struct k_heap llext_instr_heap;
+struct k_heap llext_data_heap;
+#else
 struct k_heap llext_heap;
+#endif
```

Comment:
In the context of always having "separate" data/instr heaps for the different functions, even though they may be used as a single.
Wild idea, would it be possible to do this?
```suggestion
#else
struct k_heap llext_heap;
#define llext_instr_heap llext_heap
#define llext_data_heap llext_heap
#endif
```

would this reduce the #ifdefs in the rest of the code?

Reply:
I went with Tom's suggestion, but if you prefer your suggestion, we can go with it instead since Tom said he doesn't have a strong preference

Reply:
Made the change! Yeah, it looks a lot better